design/pushed-authorization-requests.md

```diff
@@ -0,0 +1,326 @@
+
+````
+HTTP/1.1 201 Created
+Content-Type: application/json
+Cache-Control: no-cache, no-store
+
+{
+   "request_uri":
+   "urn:ietf:params:oauth:request_uri:bwc4JK-ESC0w8acc191e-Y1LTC2",
+   "expires_in": 60
+}
+````
+
+##### Example PAR Error
+````
+HTTP/1.1 400 Bad Request
+Content-Type: application/json
+Cache-Control: no-cache, no-store
+
+{
+   "error": "invalid_request",
+   "error_description": "The redirect_uri is not valid for the given client"       
+}
+````
+
+
+### Representations
+
+#### PAR Response
+
+- **request_uri**
+
+The request URI corresponding to the authorization request posted. This URI is used as reference to the respective request data in the subsequent authorization request only.
+
+The format of the "request_uri" value is at the discretion of the authorization server but it MUST contain some part generated using a cryptographically strong pseudorandom algorithm such that it is computationally infeasible to predict or guess a valid value. The authorization server MAY construct the "request_uri" value using the form "urn:ietf:params:oauth:request_uri:<reference-value>" with "<reference-value>" as the random part of the URI that references the respective authorization request data. The string representation of a UUID as a URN per [[RFC4122](https://tools.ietf.org/html/rfc4122)] is also an option for authorization servers to construct "request_uri" values. The "request_uri" value MUST be bound to the client that posted the authorization request.
+
+- **expires_in**
+
+A JSON number that represents the lifetime of the request URI in seconds as a positive integer. The request URI lifetime should be configurable (e.g., between 5 and 600 seconds)
+
+#### PAR Error
+
+-   **error** An error code which can take one of the following values :
+
+    -   **invalid_request** The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed.
+    -   **invalid_client** Client authentication failed, due to missing or invalid client credentials.
+    -   **unauthorized_client** The client is not registered for the requested `response_type`.
+    -   **unsupported_response_type** The server doesn't support the requested `response_type`.
+-   **[ error_description ]** Optional text providing additional information about the error that occurred.
+
+-   **[ error_uri ]** Optional URI for a web page with information about the error that occurred.
+
+
+### Authorization Request
+
+The client uses the request URI value to create the subsequent authorization request by directing the user-agent to make an HTTP request to the authorization server's authorization endpoint like the following:
+
+````
+GET /authorize?client_id=s6BhdRkqt3&request_uri=urn%3Aietf%3Aparams
+       %3Aoauth%3Arequest_uri%3Abwc4JK-ESC0w8acc191e-Y1LTC2 HTTP/1.1
+     Host: as.example.com
+````
+
+## Implementation details
+
+### require_pushed_authorization_requests parameter 
```

Comment:
This server metadata was defined in [PAR](https://tools.ietf.org/html/draft-ietf-oauth-par-06#section-5) while `request_uri_lifespan` is not standardized metadata, namely keycloak specific server metadata. Therefore, how about mentioning this point explicitly?